__tests__/chat.test.js

```diff
@@ -228,10 +229,16 @@ describe('Chat Methods', () => {
         })
         alice1.chat.send(channel, {body: `c${i} test`})
       }
-      await timeout(2000)
-      for (const i in channels) {
-        expect(okChecks[i]).toBe(true)
+      const allChecksOk = () => {
+        for (const i in channels) {
+          if (!okChecks[i]) {
+            return false
+          }
+        }
+        return true
       }
+      await pollFor(allChecksOk)
+      expect(allChecksOk()).toBe(true)
```

Comment:
Since we're in JS rather than iced, why not throw a timeout error in pollFor? I think that'd be more intuitive, I nearly pushed broken tests to master in iced because I didn't realize that pollFor will "fail silently". If you keep it as-is for consistency across codebases then I'll totally understand it, it's just a suggestion.

Reply:
oh, that's fine. This has nothing to do with iced. I just figured the test would fail either way if the poll timed out. But I'll change it.

Reply:
Idea #2 - make `pollFor` return the final value, so this would become

`expect(await pollFor(allChecksOk)).toBe(true)`

Still won't require duplicate checks after pollFor while making the code clearer as the syntax itself will describe what it's doing vs just the name.

Reply:
I just went with option 1 before I saw idea 2. Option 1 seems fine - this is just a little helper fn for tests anyway.